src/spreadinterp.cpp

```diff
@@ -1798,8 +1797,8 @@ auto ker_eval(FLT *FINUFFT_RESTRICT ker, const finufft_spread_opts &opts,
     }
     if constexpr (kerevalmeth == 0) {
       alignas(simd_type::arch_type::alignment()) std::array<T, MAX_NSPREAD> kernel_args{};
-      set_kernel_args(kernel_args.data(), inputs[i], opts);
-      evaluate_kernel_vector(ker + (i * MAX_NSPREAD), kernel_args.data(), opts, ns);
+      set_kernel_args<ns>(kernel_args.data(), inputs[i]);
```

Comment:
Ok, I can see why templating by ns helps here with speed. Question is, is kerevalmeth=0 supposed to be fast, or merely a readable reference? For me, it is the latter.

Reply:
I was checking the problem in the test and made some changes. I also made this change but since it does not break anything or reduce clarity I decided to keep it in. 

If someone uses custom oversampling factors this code is still needed so I thought it was worth making the change.